### PR TITLE
Handle empty expiration date and make id required in admin banner update cmd

### LINKED ISSUE #867: [Bug] admin banner update: Clear expiration date doesn't work as documented
**Describe the bug**
In the docs https://docs.microsoft.com/en-ca/cli/azure/ext/azure-devops/devops/admin/banner#ext-azure-devops-az-devops-admin-banner-update for the expiration argument it says

- "To unset the expiration for the banner, supply an empty value to this argument.".

I tried

- Not including the --expiration argument
- --expiration
- --expiration ''
- --expiration ""

as in

- az devops admin banner update --id '' --org https://dev.azure.com/MyOrgUrl/ --expiration ''

None of these worked, instead I got an error msg:

- az : ERROR: az devops admin banner update: error: argument --expiration: expected one argument

**To Reproduce**
Azure Cli Version: 
azure-cli                         2.0.75

Azure-Devops extension version: 
azure-devops                      0.13.0

1. login using az login
2. setup defaults: az devops admin banner add --message 'Sensitivity: Unclassified -- No Protected Information on this site' --org https://dev.azure.com/MyOrgUrl/ --expiration '2019-11-30 23:59:59 UTC'
3. Run command: az devops admin banner update --id '[banner-guid]' --org https://dev.azure.com/MyOrgUrl/ --expiration ''
4. See error: az : ERROR: az devops admin banner update: error: argument --expiration: expected one argument

**Expected behavior**
The banner expiration date should have changed to no expiration

**Screenshots**
n/a

**Debug logs**
PS C:\Users\[username]> az devops admin banner update --id '[banner-guid]' --org https://dev.azure.com/MyOrgUrl/ --expiration "" --debug

az : DEBUG: Command arguments: ['devops', 'admin', 'banner', 'update', '--id', '[banner-guid]', '--org', 'https://dev.azure.com/MyOrgUrl/', '--expiration', '--debug']
At line:1 char:1
+ az devops admin banner update --id '[banner-guid] ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (DEBUG: Command ...on', '--debug']:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
DEBUG: Event: Cli.PreExecute []
DEBUG: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x037770C0>, <function OutputProducer.on_global_arguments at 0x03602CD8>, <function CLIQuery.on_global_arguments at 0x03630DB0>]
DEBUG: Event: CommandInvoker.OnPreCommandTableCreate []
DEBUG: Installed command modules ['acr', 'acs', 'advisor', 'ams', 'apim', 'appconfig', 'appservice', 'backup', 'batch', 'batchai', 'billing', 'botservice', 'cdn', 'cloud', 'cognitiveservices', 'configure', 'consumption', 'container', 'cosmosdb', 'deploymentmanager', 'dla', 'dls', 'dms', 
'eventgrid', 'eventhubs', 'extension', 'feedback', 'find', 'hdinsight', 'interactive', 'iot', 'iotcentral', 'keyvault', 'kusto', 'lab', 'managedservices', 'maps', 'monitor', 'natgateway', 'netappfiles', 'network', 'policyinsights', 'privatedns', 'profile', 'rdbms', 'redis', 'relay', 'reservations', 
'resource', 'role', 'search', 'security', 'servicebus', 'servicefabric', 'signalr', 'sql', 'sqlvm', 'storage', 'vm']
DEBUG: Loaded module 'acr' in 0.007 seconds.
DEBUG: Loaded module 'acs' in 0.014 seconds.
DEBUG: Loaded module 'advisor' in 0.003 seconds.
DEBUG: Event: CommandLoader.OnLoadCommandTable []
DEBUG: Loaded module 'ams' in 0.008 seconds.
DEBUG: Loaded module 'apim' in 0.006 seconds.
DEBUG: Loaded module 'appconfig' in 0.009 seconds.
DEBUG: Loaded module 'appservice' in 0.013 seconds.
DEBUG: Loaded module 'backup' in 0.005 seconds.
DEBUG: Event: CommandLoader.OnLoadCommandTable []
DEBUG: Loaded module 'batch' in 0.013 seconds.
DEBUG: Loaded module 'batchai' in 0.006 seconds.
DEBUG: Loaded module 'billing' in 0.005 seconds.
DEBUG: Loaded module 'botservice' in 0.009 seconds.
DEBUG: Event: CommandLoader.OnLoadCommandTable []
DEBUG: Loaded module 'cdn' in 0.006 seconds.
DEBUG: Loaded module 'cloud' in 0.011 seconds.
DEBUG: Loaded module 'cognitiveservices' in 0.008 seconds.
DEBUG: Loaded module 'configure' in 0.004 seconds.
DEBUG: Loaded module 'consumption' in 0.007 seconds.
DEBUG: Loaded module 'container' in 0.005 seconds.
DEBUG: Loaded module 'cosmosdb' in 0.012 seconds.
DEBUG: Loaded module 'deploymentmanager' in 0.010 seconds.
DEBUG: Loaded module 'dla' in 0.024 seconds.
DEBUG: Loaded module 'dls' in 0.007 seconds.
DEBUG: Loaded module 'dms' in 0.005 seconds.
DEBUG: Loaded module 'eventgrid' in 0.007 seconds.
DEBUG: Loaded module 'eventhubs' in 0.010 seconds.
DEBUG: Loaded module 'extension' in 0.005 seconds.
DEBUG: Loaded module 'feedback' in 0.005 seconds.
DEBUG: Loaded module 'find' in 0.005 seconds.
DEBUG: Loaded module 'hdinsight' in 0.007 seconds.
DEBUG: Loaded module 'interactive' in 0.002 seconds.
DEBUG: Loaded module 'iot' in 0.010 seconds.
DEBUG: Loaded module 'iotcentral' in 0.004 seconds.
DEBUG: Loaded module 'keyvault' in 0.010 seconds.
DEBUG: Loaded module 'kusto' in 0.005 seconds.
DEBUG: Loaded module 'lab' in 0.007 seconds.
DEBUG: Loaded module 'managedservices' in 0.006 seconds.
DEBUG: Loaded module 'maps' in 0.006 seconds.
DEBUG: Loaded module 'monitor' in 0.009 seconds.
DEBUG: Loaded module 'natgateway' in 0.004 seconds.
DEBUG: Event: CommandLoader.OnLoadCommandTable []
DEBUG: Loaded module 'netappfiles' in 0.006 seconds.
DEBUG: Loaded module 'network' in 0.032 seconds.
DEBUG: Loaded module 'policyinsights' in 0.007 seconds.
DEBUG: Loaded module 'privatedns' in 0.008 seconds.
DEBUG: Loaded module 'profile' in 0.004 seconds.
DEBUG: Loaded module 'rdbms' in 0.009 seconds.
DEBUG: Loaded module 'redis' in 0.004 seconds.
DEBUG: Loaded module 'relay' in 0.006 seconds.
DEBUG: Loaded module 'reservations' in 0.005 seconds.
DEBUG: Loaded module 'resource' in 0.009 seconds.
DEBUG: Loaded module 'role' in 0.007 seconds.
DEBUG: Loaded module 'search' in 0.006 seconds.
DEBUG: Loaded module 'security' in 0.005 seconds.
DEBUG: Loaded module 'servicebus' in 0.008 seconds.
DEBUG: Loaded module 'servicefabric' in 0.004 seconds.
DEBUG: Loaded module 'signalr' in 0.004 seconds.
DEBUG: Loaded module 'sql' in 0.013 seconds.
DEBUG: Loaded module 'sqlvm' in 0.005 seconds.
DEBUG: Event: CommandLoader.OnLoadCommandTable []
DEBUG: Loaded module 'storage' in 0.037 seconds.
DEBUG: Loaded module 'vm' in 0.021 seconds.
DEBUG: Loaded all modules in 0.501 seconds. (note: there's always an overhead with the first module loaded)
DEBUG: Extensions directory: 'C:\Users\[username]\.azure\cliextensions'
DEBUG: Found 2 extensions: ['aks-preview', 'azure-devops']
DEBUG: Extensions directory: 'C:\Users\[username]\.azure\cliextensions'
DEBUG: Extension compatibility result: is_compatible=True cli_core_version=2.0.75 min_required=2.0.49 max_required=None
DEBUG: Event: CommandLoader.OnLoadCommandTable []
DEBUG: Extensions directory: 'C:\Users\[username]\.azure\cliextensions'
DEBUG: Loaded extension 'aks-preview' in 0.008 seconds.
DEBUG: Extensions directory: 'C:\Users\[username]\.azure\cliextensions'
DEBUG: Extension compatibility result: is_compatible=True cli_core_version=2.0.75 min_required=2.0.69 max_required=None
DEBUG: Extensions directory: 'C:\Users\[username]\.azure\cliextensions'
DEBUG: Loaded extension 'azure-devops' in 0.053 seconds.
DEBUG: Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x037C3348>]
INFO: az_command_data_logger : command args: devops admin banner update --id {} --org {} --expiration --debug
DEBUG: metadata file logging enabled - writing logs to 'C:\Users\[username]\.azure\commands'.
DEBUG: Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x038B57C8>]
DEBUG: Event: CommandInvoker.OnPostArgumentLoad []
DEBUG: Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x038B5780>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x038B5E40>]
DEBUG: Event: CommandInvoker.OnCommandTableLoaded []
DEBUG: Event: CommandInvoker.OnPreParseArgs [<function _documentdb_deprecate at 0x045BA588>]
ERROR: cli.azure.cli.core.parser : az devops admin banner update: error: argument --expiration: expected one argument
ERROR: az devops admin banner update: error: argument --expiration: expected one argument
usage: az devops admin banner update [-h] [--verbose] [--debug]
                                     [--output {json,jsonc,table,tsv,yaml,none}]
                                     [--query JMESPATH]
                                     [--subscription _SUBSCRIPTION]
                                     [-m MESSAGE] [-t {info,warning,error}]
                                     [--id ID] [--expiration EXPIRATION]
                                     [--organization ORGANIZATION]
                                     [--detect [{true,false}]]
INFO: az_command_data_logger : exit code: 2
INFO: telemetry.save : Save telemetry record of length 2555 in cache
WARNING: telemetry.check : Negative: The C:\Users\[username]\.azure\telemetry.txt was modified at 2019-11-04 09:40:22.142383, which in less than 600.000000 s
INFO: command ran in 1.029 seconds.

**Additional context**
First raised as doc feedback issue, [https://github.com/Azure/azure-cli-extensions/issues/1053](https://github.com/Azure/azure-cli-extensions/issues/1053)

### PR DESCRIPTION
**Pull request by ishitam8**
Please make sure the code is following contribution guidelines in CONTRIBUTING.md

 - [ ] : This PR has a corresponding issue open in the Repository.
 - [ ] : Approach is signed off on the issue.

Fixes #867

### CODE REVIEW

File: azure-devops/azext_devops/dev/admin/banner.py [hunk 1]
```diff
@@ -82,7 +82,10 @@ def banner_update(message=None, banner_type=None, id=None, expiration=None, orga
         raise ValueError('At least one of the following arguments need to be supplied: --message, --type, '
                          '--expiration.')
     if expiration is not None:
```
Comment (gauravsaralMs) [inline azure-devops/azext_devops/dev/admin/banner.py]: something like this?
``` python
if expiration is not None and expiration != '':
    expiration_iso8601 = convert_date_string_to_iso8601(value=expiration, argument='expiration')
else:
    expiration_iso8601 = expiration
```

btw: is None and `` behave differently while calling API?

Comment (ishitam8) [reply azure-devops/azext_devops/dev/admin/banner.py]: Yes, to unset the date we explicitly have to pass empty string [i.e ''] and not None.
In case of none, expiration date will remain unchanged.

Comment (atbagga) [reply azure-devops/azext_devops/dev/admin/banner.py]: For future reference.. More pythonic way of testing a string as none or empty.

```
if expiration: 
```
